File: ViewPagerWithDynamicViews/app/src/main/java/com/example/viewpagerwithdynamicviews/Views/FirstFragment.java
package com.example.viewpagerwithdynamicviews.Views;

import android.os.Bundle;

import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.example.viewpagerwithdynamicviews.R;

public class FirstFragment extends Fragment {


    public FirstFragment() {
        // Required empty public constructor
    }


    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) { 
        return inflater.inflate(R.layout.fragment_first, container, false);
    }
}

File: KotlinViewModelExample/app/src/main/java/com/example/kotlinviewmodelexample/FirstFragment.kt
package com.example.kotlinviewmodelexample


import android.arch.lifecycle.Observer
import android.arch.lifecycle.ViewModelProviders
import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v4.app.FragmentActivity
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView


// TODO: Rename parameter arguments, choose names that match
// the fragment initialization parameters, e.g. ARG_ITEM_NUMBER
private const val ARG_PARAM1 = "param1"
private const val ARG_PARAM2 = "param2"

/**
 * A simple [Fragment] subclass.
 *
 */
class FirstFragment : Fragment() {
    private lateinit var mModel:DummyViewModel
    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        // Inflate the layout for this fragment
        mModel = ViewModelProviders.of(activity as FragmentActivity).get(DummyViewModel::class.java)
        return inflater.inflate(R.layout.fragment_first, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        var textView = view.findViewById(R.id.randValue) as TextView
        mModel.setRandomNumber.observe(this, Observer { data->
            textView.text = "DATA : $data"
        })
        super.onViewCreated(view, savedInstanceState)
    }

}
